### DISCUSSION TITLE
CS2-Ultimate-Mapchooser

### DISCUSSION THREAD

**Discussion by 4fc7c8f789** [kus/cs2-modded-server#68] [Q&A] [answered]
This set-up is using a undocumented plugin called CS2-Ultimate-Mapchooser. It has a unique functionality I could not find in comparable plugins, in that it initiates votes based on the currently active mapgroup and not based on a pre-defined maps.txt file.

I cannot find any further details about the mod online. Can someone point me in the right direction for install instructions, or suggest an alternative?

**Comment by kus** [comment, marked as answer]
It’s an unreleased plugin I wrote.

You can take the .dll from [here](https://github.com/kus/cs2-modded-server/tree/master/game/csgo/addons/counterstrikesharp/plugins/CS2-Ultimate-Mapchooser) and put it on your server.

The only thing you need to do is when you set `mapgroup` you also set it with `css_mapgroup` which corresponds to your [gamemodes_server.txt](https://github.com/kus/cs2-modded-server/blob/master/game/csgo/gamemodes_server.txt) file.

i.e.:
```
mapgroup mg_dm
css_mapgroup mg_dm
```

It chooses a random map from the current mapgroup at the end of the map.

You can also use `!map` in chat to change map to a map in the mapgroup as an admin.

**Reply by 4fc7c8f789** [reply]
Thanks a lot! I've tried that approach but for some reason the plugin is not getting loaded on startup (according to the terminal). The server is also not reacting to the !map command.

Having said that, I did not declare my mapgroup the way you are describing, so will retry...

Let me know if you can think of another reason I managed to screw it up :)

Appreciate the great work on the server set-up. This directory helped me greatly to onboard on configuring my first server.

**Reply by 4fc7c8f789** [reply]
Confirming that the approach works. Thanks!

Would love to see an evolution of this plugin at some point (for stuff like !rtv, !nominate) :)
